ts/src/bingx.ts

```diff
@@ -2218,6 +2218,7 @@ export default class bingx extends Exchange {
         const types = {
             'trigger_market': 'market',
             'trigger_limit': 'limit',
+            'stop_limit': 'limit',
```

Comment:
@carlosmiei You may want to add stop_market as type as well ;) that works essentially identically.
```
{'info': {'symbol': 'ETH-USDT',
  'orderId': '1792461196318638080',
  'price': '3094.21',
  'StopPrice': '2778.42',
  'origQty': '0.032359',
  'executedQty': '0',
  'cummulativeQuoteQty': '0',
  'status': 'FAILED',
  'type': 'TAKE_STOP_MARKET',
  'side': 'SELL',
  'time': '1716191026000',
  'updateTime': '1716191094000',
  'origQuoteOrderQty': '0',
  'clientOrderID': '',
  'fee': '0'},
 'id': '1792461196318638080',
 'clientOrderId': None,
 'symbol': 'ETH/USDT',
 'timestamp': 1716191026000,
 'datetime': '2024-05-20T07:43:46.000Z',
 'lastTradeTimestamp': 1716191094000,
 'lastUpdateTimestamp': 1716191094000,
 'type': 'take_stop_market',
 'timeInForce': None,
 'postOnly': None,
 'side': 'sell',
 'price': 3094.21,
 'stopPrice': None,
 'triggerPrice': None,
 'stopLossPrice': None,
 'takeProfitPrice': None,
 'average': None,
 'cost': 0.0,
 'amount': 0.032359,
 'filled': 0.0,
 'remaining': 0.032359,
 'status': 'canceled',
 'fee': {'currency': 'USDT', 'cost': '0'},
 'trades': [],
 'reduceOnly': None,
 'fees': [{'currency': 'USDT', 'cost': 0.0}]}
 ```

Reply:
yes @xmatthias I didn't add the -market types like `stop_market` because that would require a deeper fix inside `safeOrder` (right now if type=market infers TIF = IOC which is not the case ), but you're right  we should probably tackle it now